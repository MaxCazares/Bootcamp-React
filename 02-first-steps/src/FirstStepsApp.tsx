import { ItemCounter } from "./shopping-cart/ItemCounter";

interface ItemInCart {
	product_name: string;
	quantity: number;
}

const ItemsInCart: ItemInCart[] = [
	{ product_name: 'Cuerdas para bajo', quantity: 5 },
	{ product_name: 'Pedal Fuzz para bajo', quantity: 3 },
	{ product_name: 'Funda semirigida para bajo', quantity: 6 },
]

export const FirstStepsApp = () => {
	return (
		<div>
			{
				ItemsInCart.map(({ product_name, quantity }) => (
					<ItemCounter key={product_name} product_name={product_name} quantity={quantity} />
				))
			}
		</div>
	);
};
