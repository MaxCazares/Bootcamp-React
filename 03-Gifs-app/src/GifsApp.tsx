import { CustomHeader } from './Components/CustomHeader';
import { GifList } from './Components/Gifs/GifList';
import { PreviousSearches } from './Components/Gifs/PreviousSearches';
import { SearchBar } from './Components/Gifs/SearchBar';
import { useGif } from './Hooks/useGif';


export const GifsApp = () => {
	const {gifs, previousTerms, handleSearch, handleTermClicked} = useGif()

	return (
		<>
			<CustomHeader
				title='Buscador de Gifs'
				description='Busca y comparte el gif perfecto'
			/>

			<SearchBar
				placeholder='Busca lo que mÃ¡s te gusta'
				onQuery={handleSearch}
			/>

			<PreviousSearches
				previousSearches={previousTerms}
				onLabelClicked={handleTermClicked}
			/>

			<GifList gifs={gifs} />
		</>
	);
};
