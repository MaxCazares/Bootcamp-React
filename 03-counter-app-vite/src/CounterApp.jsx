import { Fragment, useState } from "react"

export const CounterApp = ({value = 0}) => {

    const [counter, setCounter] = useState(value)

    const handleSubstract = () => setCounter(counter - 1)
    const handleReset = () => setCounter(value)
    const handleAdd = () => setCounter(counter + 1)

    return (
        <Fragment>
            <h1>Counter App</h1>
            <h2 data-testid='test-counter'>{counter}</h2>
            <button aria-label="btn-decrease" onClick={handleSubstract}>-1</button>
            <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
            <button aria-label="btn-increase" onClick={handleAdd}>+1</button>
        </Fragment>
    )
}