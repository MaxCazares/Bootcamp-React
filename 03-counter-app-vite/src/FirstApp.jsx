import { Fragment } from "react"
import PropTypes from 'prop-types'

const person = {
    'Name': 'Max',
    'Age': 25
}

const getFactorialNumber = number => {
    var factorial = 1;
    for (let i = 1; i <= number; i++) 
        factorial *= i;

    return factorial
}

export const FirstApp = ({title = 'React', factorial = 1}) => {
    return (
    <Fragment>
        <h2>{title}'s First App</h2>
        <code>{JSON.stringify(person)}</code>
        <p>{getFactorialNumber(factorial)}</p>
    </Fragment>
    )
}
