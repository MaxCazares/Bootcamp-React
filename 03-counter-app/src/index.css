html, body{
    background-color: #21232A;
    color: #fff;
    font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
    font-size: 13.rem;
    padding: 70px;
}

button{
    font-size: 1.3rem;
    padding: 10px;
    margin-right: 10px;
}