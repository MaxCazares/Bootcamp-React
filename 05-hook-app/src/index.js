import React from 'react';
import ReactDOM from 'react-dom';
// import CounterApp from './components/01-useState/CounterApp';
// import HookApp from './HookApp';
// import CounterCustom from './components/01-useState/CounterCustom';
// import SimpleForm from './components/02-useEffect/SimpleForm';
// import FormCustom from './components/02-useEffect/FormCustom';
// import MultipleCustomHooks from './components/03-examples/MultipleCustomHooks';
// import FocusScreen from './components/04-useRef/FocusScreen';
// import RealExampleRef from './components/04-useRef/RealExampleRef';
// import LayoutEffect from './components/05-useLayoutEffect/LayoutEffect';
// import Memorize from './components/Memos/Memorize';
// import MemoHook from './components/Memos/MemoHook';
// import CallbackHook from './components/06-memos/CallbackHook';
// import { Padre } from './components/07-tarea-memo/Padre';
// import ToDoApp from './components/08-useReducer/ToDoApp';
import MainApp from './components/09-useContext/MainApp';



ReactDOM.render(
    <React.StrictMode>
        <MainApp />
    </React.StrictMode>,
    document.querySelector('#root')
);

// import './components/08-useReducer/intro-reducer'