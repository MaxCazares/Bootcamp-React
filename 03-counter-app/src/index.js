import React from 'react';
import ReactDOM from 'react-dom';
// import PrimeraApp from './PrimeraApp';
import CounterApp from './CounterApp';
import './index.css'

const root = document.querySelector('#root');

// ReactDOM.render(<PrimeraApp saludo='Hola Mundo' />, root);
ReactDOM.render(<CounterApp value={123} />, root);