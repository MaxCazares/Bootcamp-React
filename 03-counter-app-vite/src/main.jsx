import React from "react";
import ReactDOM from "react-dom/client";

import './styles.css'

import {App} from './HelloWorldApp.jsx'     
import {FirstApp} from './FirstApp.jsx'
import {CounterApp} from './CounterApp.jsx'


ReactDOM.createRoot((document.getElementById('root'))).render(
    <React.StrictMode>
        {/* <App/> */}
        {/* <FirstApp title="Max" factorial={10}/> */}
        <CounterApp value={123}/>
    </React.StrictMode>
)

