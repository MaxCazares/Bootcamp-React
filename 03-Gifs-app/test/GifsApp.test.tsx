import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';

import { GifsApp } from '../src/GifsApp';

describe('GifsApp', () => {
	test('Should match with Snapshot', () => {
		const { container } = render(<GifsApp />);
		expect(container).toMatchSnapshot();
	});
});
