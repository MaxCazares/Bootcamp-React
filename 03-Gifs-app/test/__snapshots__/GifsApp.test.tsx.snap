// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifsApp > Should match with Snapshot 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Buscador de Gifs
    </h1>
    <p>
      Busca y comparte el gif perfecto
    </p>
  </div>
  <div
    class="search-container"
  >
    <input
      placeholder="Busca lo que mÃ¡s te gusta"
      type="text"
      value=""
    />
    <button>
      Buscar
    </button>
  </div>
  <div
    class="previous-searches"
  >
    <ul
      class="previous-searches-list"
    />
  </div>
  <div
    class="gifs-container"
  />
</div>
`;
